openseespy-pip/openseespy/postprocessing/Get_Rendering.py

```diff
@@ -1448,3 +1376,254 @@ def update_plot(ii):
     plt.show()
     return line_ani
 
+
+
+
+
+# =============================================================================
+# Experimental Functions
+# =============================================================================
+
+def _plot_on_click(self, event):
+    
+    timeSteps = self.timeSteps
+    Disp_nodeArray =  self.Disp_nodeArray
+    self.DeflectedNodeCoordArray
+    
+    # Get Node
+    
+    # plot Nodes
+    
+    fig, ax = plt.subplots()
+    plt.plot(timeSteps)
+    
+    
+    
+    thisline = event.artist
+    xdata = thisline.get_xdata()
+    ydata = thisline.get_ydata()
+    ind = event.ind
+    points = tuple(zip(xdata[ind], ydata[ind]))
+    print('onpick points:', points)
+
+    
+class plot_deformedshape_Events:
+    
+    def __init__(self, Model="none", LoadCase="none", tstep = -1, scale = 10, overlap='no'):
+        self.Model = Model
+        self.LoadCase = LoadCase
+        self.tstep = tstep
+        self.scale = scale
+        self.overlap = overlap
+        
+        self.timeSteps, self.Disp_nodeArray = idbf._readNodeDispData(self.Model, self.LoadCase)
+        self.nodeArray, self.elementArray = idbf._readNodesandElements(self.Model)
+
+    def _plot_on_click(self, event):
```

Comment:
I see this works great for a 2D model where it plots x-disp on clicking a node. What does it do for a 3D model? Is it possible to extend this with subplots for each dof? x-disp and y-disp for a 2D model and x, y and z-disp for a 3D model.

Reply:
It is possible to extend to 3D. If this is something we like, I will do that.

Reply:
This is a very cool function to have. I like it.